scripts/source-maintenance.sh

```diff
@@ -493,6 +493,27 @@ make -C src/http gperf-files
 
 run_ checkMakeNamedErrorDetails || exit 1
 
+# extract Author details from commit log
+collectAuthors ()
+{
+  tail -n +5 CONTRIBUTORS | sed -e 's/^[ ]*//' >AUTHORS.tmp
+
+  git log |
+    grep -E "^[ ]*[aA]uthor:" |
+    sed -e 's/^[ \t]*//' -e 's/[Aa]uthor:[ ]*//' -e 's/"//g' |
+    sed -e 's/[ ]*[,+&] /\n/g' >>AUTHORS.tmp
+
+  (
+    head -n 4 CONTRIBUTORS
+    sort -uf AUTHORS.tmp | while read line ; do echo "    $line" ; done
+  )>CONTRIBUTORS.new
```

Comment:
> Alex: This approach precludes us from listing contributors not mentioned in git log

No, I was wrong. The script does populate `AUTHORS.tmp` with entries from the existing `CONTRIBUTORS` file. Sorry.

> Perhaps more importantly, prior applications of ~this or~ similar script have resulted in contributor losses

That part is essentially correct, but we will not have that problem going forward, which is a good thing. We just need to restore the already lost entries. Once that bootstrapping is done, the restored entries will be preserved.

Reply:
Good find. IIRC that removal was because their contribution was the script which is no longer published. Makes sense to restore the entry though since we are using the whole commit history for CONTRIBUTORS.